java/src/main/java/io/cucumber/cienvironment/CiEnvironmentImpl.java

```diff
@@ -83,17 +84,11 @@ private String evaluateRevision(Map<String, String> env) {
             if (env.get("GITHUB_EVENT_PATH") == null) {
                 throw new RuntimeException("GITHUB_EVENT_PATH not set");
             }
-            Path path = Paths.get(env.get("GITHUB_EVENT_PATH"));
             try {
-                String json = Files.lines(path).collect(Collectors.joining());
-                Json ob = Json.read(json);
-                try {
-                    return ob.asJsonMap().get("pull_request").asJsonMap().get("head").asJsonMap().get("sha").asString();
-                } catch (RuntimeException e) {
-                    throw new RuntimeException(String.format("Could not find .pull_request.head.sha in %s:\n%s", path, json));
-                }
-            } catch (IOException e) {
-                throw new RuntimeException("Could not read " + path, e);
+                JsonValue event = Json.parse(new InputStreamReader(new FileInputStream(env.get("GITHUB_EVENT_PATH")), StandardCharsets.UTF_8));
+                return event.asObject().get("pull_request").asObject().get("head").asObject().get("sha").asString();
+            } catch (Exception e) {
+                throw new RuntimeException("Could not read .pull_request.head.sha from " + env.get("GITHUB_EVENT_PATH"), e);
```

Comment:
Does `Json.parse` not throw something more specific?

Reply:
* `Json.parse` throws the following:
  * `java.io.IOException` (the only declared exception)
  * `java.lang.NullPointerException` (undeclared)
  * `java.lang.IllegalArgumentException` (undeclared)
  * `com.eclipsesource.json.ParseException` (undeclared)
* `JsonValue.asObject` throws the following:
  * `UnsupportedOperationException` (undeclared)
* `JsonValue.get` throws the following:
  * `java.lang.NullPointerException` (undeclared)
  * `java.lang.ClassCastException` (undeclared)
* `JsonValue.asString` throws the following:
  * `UnsupportedOperationException` (undeclared)

I really can't be bothered to handle all of these independenly, which is why I have a single `catch(Exception)` clause.

Reply:
> I really can't be bothered to handle all of these independenly, which is why I have a single catch(Exception) clause.

You probably should. It would be annoying if at any time in the future cucumber stops to work because github changed it's api.

Reply:
I'm not worried about it. The original error is thrown in the `RuntimeException` and that should provide enough context about what went wrong.

Reply:
Yes. And then that version of Cucumber will never work on GitHub.

It should fail by omitting the information, not by throwing an exception.